src/Series.cpp

```diff
@@ -64,18 +64,33 @@ std::string cleanFilename(std::string const& filename, Format f);
 std::function< bool(std::string const&) > matcher(std::string const& name, Format f);
 
 #if openPMD_HAVE_MPI
-Series::Series(std::string const& filepath,
+Series::Series(std::string filepath,
                AccessType at,
                MPI_Comm comm)
         : iterations{Container< Iteration, uint64_t >()},
           m_iterationEncoding{std::make_shared< IterationEncoding >()}
 {
+#ifdef _WIN32
+    if( auxiliary::contains(filepath, '/') )
+    {
+        std::cerr << "Filepaths on WINDOWS platforms may not contain slashes '/'! "
+                  << "Replacing with backslashes '\\' unconditionally!" << std::endl;
+        filepath = auxiliary::replace_all(filepath, "\\", "/");
+    }
+#else
+    if( auxiliary::contains(filepath, '\\') )
+    {
+        std::cerr << "Filepaths on UNIX platforms may not include backslashes '\\'! "
+                  << "Replacing with slashes '/' unconditionally!" << std::endl;
+        filepath = auxiliary::replace_all(filepath, "\\", "/");
+    }
+#endif
     std::string newPath;
     std::string newName;
-    auto const pos = filepath.find_last_of('/');
+    auto const pos = filepath.find_last_of(auxiliary::directory_separator);
     if( std::string::npos == pos )
     {
-        newPath = "./";
+        newPath = "." + auxiliary::directory_separator;
```

Comment:
Clang 5 warns:
```
'const char' to a string does not append to the string
(use array indexing to silence this warning)
```

Reply:
I made that exact [change](https://github.com/C0nsultant/libopenPMD/blob/0b97c08c0094bc52f461d15d3d905e43b7ac44a6/src/Series.cpp#L173-L174) in the non-MPI constructor, but forgot to copy-paste...

The constructor looks like a good candidate for refactoring, anyway.